rasa/core/policies/rule_policy.py

```diff
@@ -270,6 +277,137 @@ def _check_rule_restriction(
                 f"Please use stories for these cases."
             )
 
+    @staticmethod
+    def _check_slots_fingerprint(
+        fingerprint: Dict[Text, List[Text]], state: State, featurized_slots: Set[Text]
+    ) -> Set[Text]:
+        # leave only featurized slots
+        expected_slots = set(fingerprint.get(ACTION_FINGERPRINT_SLOTS)).intersection(
```

Comment:
I think this should be done as part of the fingerprinting

Reply:
fingerprinting is an old method, I don't know what effects of changing it will be

Reply:
also I'd argue that fingerprints are independent of featurization, it's just in this concrete case we want to remove unfeaturized slots

Reply:
Why are the fingerprints independent? They are used to check if stuff was already seen? As unfeaturized things don't matter I think it's fine to filter that out right away